binding.cc

```diff
@@ -1107,6 +1197,77 @@ namespace zmq {
     BufferReference* bufref_;
   };
 
+  NAN_METHOD(Socket::Sendv) {
+    Socket* socket = GetSocket(info);
+    if (socket->state_ != STATE_READY)
+      return info.GetReturnValue().Set(Nan::False());
+
+    int events;
+    size_t events_size = sizeof(events);
+
+    int rc;
+
+    while (true) {
+      rc = zmq_getsockopt(socket->socket_, ZMQ_EVENTS, &events, &events_size);
+      if (rc != 0) {
+        if (zmq_errno()==EINTR) {
+          continue;
+        }
+        return Nan::ThrowError(ErrorMessage());
+      }
+      break;
+    }
+
+    if ((events & ZMQ_POLLOUT) == 0) {
+      return info.GetReturnValue().Set(Nan::False());
+    }
+
+    Local<Array> batch = info[0].As<Array>();
+    size_t len = batch->Length();
+
+    if (len == 0)
+      return info.GetReturnValue().Set(Nan::True());
+
+    if (len % 2 != 0)
+      return Nan::ThrowTypeError("Batch length must be even!");
+
+    for (size_t i = 0; i < len; i += 2) {
+      Local<Object> buf = batch->Get(i).As<Object>();;
+      Local<Number> flagsObj = batch->Get(i + 1).As<Number>();;
+
+      int flags = Nan::To<int>(flagsObj).FromJust();
+      size_t len = Buffer::Length(buf);
+
+      zmq_msg_t msg;
+      rc = zmq_msg_init_size(&msg, len);
+      if (rc != 0)
+        return Nan::ThrowError(ErrorMessage());
+
+      char * cp = static_cast<char *>(zmq_msg_data(&msg));
+      const char * dat = Buffer::Data(buf);
+      std::copy(dat, dat + len, cp);
```

Comment:
When it comes time for step 2, you can get this to work by setting the memory address of your zmq buf to a node::Buffer